### PR TITLE
Multithread in a few HLT related plugins from the EventFilter packages

### PR DESCRIPTION
**Pull request by perrotta**
Following the request in https://hypernews.cern.ch/HyperNews/CMS/get/hlt/3701.html , the following modules in the EventFilter packages have been migrated to the use of the parameterSetDescription:
- EvFFEDSelector from EventFilter/FEDInterface
- EcalRawToDigi from EventFilter/EcalRawToDigi
- ESRawToDigi from EventFilter/ESRawToDigi
- HcalRawToDigi from EventFilter/HcalRawToDigi
- GctRawToDigi from EventFilter/GctRawToDigi
- L1GlobalTriggerRawToDigi from EventFilter/L1GlobalTriggerRawToDigi
- CSCDCCUnpacker from EventFilter/CSCRawToDigi
- DTUnpackingModule from EventFilter/DTRawToDigi
- RPCUnpackingModule from EventFilter/RPCRawToDigi
- ScalersRawToDigi from EventFilter/ScalersRawToDigi
- SiPixelRawToDigi from EventFilter/SiPixelRawToDigi
- SiStripExcludedFEDListProducer from EventFilter/SiStripRawToDigi

The following two were also modified in order to become multithread friend stream modules;
- HcalRawToDigi from EventFilter/HcalRawToDigi
- DTUnpackingModule from EventFilter/DTRawToDigi

While moving to the parameterSetDescription usually the autogenerated python configuration are some general one, with a general name which did not require removing the original explicit python config in the release. The only exceptions in this PR are:
-  EventFilter/ESRawToDigi/python/esRawToDigi_cfi.py
- EventFilter/FEDInterface/python/EvFFEDSelector_cfi.py

which are both going to be removed from the release.

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
Hi @perrotta 
Please do remove the explicit cfi files from the python directories; 
this duplication will cause maintenance problems - if other cfi/cff files
in the release refer to them via include/import, those cfi/cff should be
updated to refer to the autogenerated ones instead.

**Comment by perrotta** [comment]
Hi @Martin-Grunewald 

As I wrote in the initial description of this PR, I removed the explicit cfi files in the two packages where there was no ambiguity (i.e., the explicit cfi originally in the release had the same name as the module itself). In the other cases, we do need to maintain the explicit cfi's because they are included with their specific name and configuration in  the release (and in some cases there were also more than one versions of them, meant to different usages). Of course, one could rewrite all them via include/import of the auto-generated configs: this is indeed what I did in this last commit for the CSC's.

Do you think it is really needed doing the same for all other (nine) packages? There are plenty of examples in CMSSW of explicit cfi left for plugin modules even when a configurationDescription is allowed. Even our HLT menu is full of them...

Therefore: if you really believe it is necessary to modify all such cfi's via include/import of the autogenerated one I will, but with low priority in the next days. (I can close this PR, adapt it from time to time, and reopen it when needed, in that case). Otherwise, we can let the PR as it is now, and possibly adapt the cfi's only later on.

**Comment by Martin-Grunewald** [comment]
Yes, we should, with low priority: the reason is that things will fall out of sync,
so importing/cloning the autogenerated ones is the way to go. But I also agree
it is of very low priority given the list of other (JIRA) things we have to do...

**Comment by perrotta** [comment]
All explicit cfi configurations in the packages that contain the migrated plugins have been 
updated in order to refer to the autogenerated ones, as requested by Martin.

Of course, there are plenty of other different explicit configurations of those unpackers in other packages (mostly DQM): I do not touch them in this PR.

As far as I am concerned, this PR is now complete: please L1, daq and Reco feel free of evaluating and possibly sign it.

**Comment by slava77** [comment]
please test

[hoping it's triggered automatically from this message]

File: EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc [hunk 1]
```diff
@@ -59,6 +61,25 @@ HcalRawToDigi::HcalRawToDigi(edm::ParameterSet const& conf):
...
+  edm::ParameterSetDescription desc;
+  desc.addUntracked<int>("HcalFirstFED",int(FEDNumbering::MINHCALFEDID));
+  desc.add<int>("firstSample",0);
+  desc.add<int>("lastSample",9);
+  desc.add<bool>("FilterDataQuality",true);
+  desc.addUntracked<std::vector<int>>("FEDs", std::vector<int>());
+  desc.addUntracked<bool>("UnpackZDC",true);
+  desc.addUntracked<bool>("UnpackCalib",true);
+  desc.addUntracked<bool>("UnpackTTP",true);
+  desc.addUntracked<bool>("silent",false);
```
Comment (slava77) [inline EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc]: this looks like a bug
https://github.com/perrotta/cmssw/blob/multithreadEventFilter/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc#L24
has
```C++
 silent_(conf.getUntrackedParameter<bool>("silent",true)),
```

**Comment by slava77** [comment]
-1

for https://github.com/cms-sw/cmssw/pull/7029	 85ef516

need a fix in HcalRawToDigi

I checked expanded configs for MC and data and found only one inconsistency https://github.com/cms-sw/cmssw/pull/7029#discussion_r22629149

This can become quite noisy.

File: EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc (hunk 1 above)
Comment (perrotta) [inline EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc]: Indeed,,, Thank you @slava77 for catching it.
Fixed now.

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/7029 5e9fe92

tested in CMSSW_7_4_X_2015-01-07-1000 (test area sign483)

short matrix outputs are now the same (prev iteration had diffs in logwarnings).
As I mentioned above, I also checked expanded configs in step2 and step3 for data (with re-HLT) and MC: the remaining changes in parameters are as expected.

**Comment by slava77** [comment]
please test

I'm curious if cms-bot would listen to me ;)

**Comment by slava77** [comment]
it actually did. uhm. was it because Andrea posted the "please test" before the commit update and signature update?
@ktf ?

**Comment by perrotta** [comment]
Feeling jealous...   ;)


Slava Krutelyov <notifications@github.com> ha scritto:

**Comment by davidlange6** [comment]
bypassing daq/l1 given simple changes. Please complain if not ok.

**Pull request merged**